packages/sources/nft-blue-chip/src/transport/marketcap.ts

```diff
@@ -0,0 +1,287 @@
+  },
+  {
+    name: 'Bored Ape Yacht Club',
+    feed: '0x352f2Bc3039429fC2fe62004a1575aE74001CfcE',
+    token: '0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D',
+  },
+  {
+    name: 'CloneX',
+    feed: '0x021264d59DAbD26E7506Ee7278407891Bb8CDCCc',
+    token: '0x49cF6f5d44E70224e2E23fDcdd2C053F30aDA28B',
+  },
+  {
+    name: 'CoolCats',
+    feed: '0xF49f8F5b931B0e4B4246E4CcA7cD2083997Aa83d',
+    token: '0x1A92f7381B9F03921564a437210bB9396471050C',
+  },
+  {
+    name: 'Cryptoadz',
+    feed: '0xFaA8F6073845DBe5627dAA3208F78A3043F99bcA',
+    token: '0x1CB1A5e65610AEFF2551A50f76a87a7d3fB649C6',
+  },
+  {
+    name: 'CryptoPunks',
+    feed: '0x01B6710B01cF3dd8Ae64243097d91aFb03728Fdd',
+    token: '0xb47e3cd837dDF8e4c57F05d70Ab865de6e193BBB',
+  },
+  {
+    name: 'Doodles',
+    feed: '0x027828052840a43Cc2D0187BcfA6e3D6AcE60336',
+    token: '0x8a90CAb2b38dba80c64b7734e58Ee1dB38B8992e',
+  },
+  {
+    name: 'Mutant Ape Yacht Club',
+    feed: '0x1823C89715Fe3fB96A24d11c917aCA918894A090',
+    token: '0x60E4d786628Fea6478F785A6d7e704777c86a7c6',
+  },
+  {
+    name: 'VeeFriends',
+    feed: '0x35bf6767577091E7f04707c0290b3f889e968307',
+    token: '0xa3AEe8BcE55BEeA1951EF834b99f3Ac60d1ABeeB',
+  },
+  {
+    name: 'World of Women',
+    feed: '0xDdf0B85C600DAF9e308AFed9F597ACA212354764',
+    token: '0xe785E82358879F061BC3dcAC6f0444462D4b5330',
+  },
+]
+
+export type MarketcapTransportGenerics = TransportGenerics & {
+  Provider: {
+    RequestBody: unknown
+    ResponseBody: unknown
+  }
+  CustomSettings: typeof customSettings
+}
+
+export interface MarketcapTransportConfig {
+  options: {
+    requestCoalescing: {
+      enabled: boolean
+      entropyMax?: number
+    }
+  }
+}
+
+export class MarketcapTransport implements Transport<MarketcapTransportGenerics> {
+  inFlightPrefix!: string
+  cache!: Cache<AdapterResponse<MarketcapTransportGenerics['Response']>>
+  inFlightCache!: Cache<boolean>
+  rateLimiter!: RequestRateLimiter
+
+  constructor(protected config: MarketcapTransportConfig) {}
+
+  async initialize(
+    dependencies: AdapterDependencies,
+    config: AdapterConfig<MarketcapTransportGenerics['CustomSettings']>,
+  ): Promise<void> {
+    this.inFlightPrefix = 'InFlight-'
+    this.cache = dependencies.cache as Cache<
+      AdapterResponse<MarketcapTransportGenerics['Response']>
+    >
+    this.inFlightCache = dependencies.cache as Cache<boolean>
+    this.rateLimiter = dependencies.requestRateLimiter
+
+    this.config.options.requestCoalescing.enabled = config.REQUEST_COALESCING_ENABLED
+    this.config.options.requestCoalescing.entropyMax = config.REQUEST_COALESCING_ENTROPY_MAX
+  }
+
+  protected async waitUntilUnderRateLimit(
+    options: {
+      maxRetries: number
+      msBetweenRetries: number
+    },
+    retry = 0,
+  ): Promise<void> {
+    if (this.rateLimiter.isUnderLimits()) {
+      logger.trace('Incoming request would not be under limits, moving on')
+      return
+    }
```

Comment:
A lot of the transport code is boilerplate copied from the HTTPTransport in `external-adapters-framework-js`. The main custom code starts with the collection data fetching (around line 194) and finishes once the total market cap in USD is found.